### PR TITLE
Unread messages amount

### PR DESCRIPTION
**Pull request by BigBy996**

### CODE REVIEW

File: features/page/homePage.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ const homePageSelector = '.nH.bkL';
 const messagesHeaderSelector = 'div[id=":3v"]';
 const totalMessagesHeaderSelector = 'span > span:nth-child(2)';
 const messageCommonSelector = '.PF.xY';
+const inboxLeftSelector = '.TN.GLujEb.aHS-bnt';
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please rename

File: features/page/homePage.js [hunk 2]
```diff
@@ -15,6 +18,22 @@ const homePage = Object.create({
...
+      if(numIndex != -1) {
+          return Number(text.substr(numIndex, text.search('\\)') - numIndex));
+      }
+      else {
+          return 0;
+      }
+  },
+
+  inboxUnreadAmountValue: function inboxUnreadAmountValue(){
+      return browser.$$(inboxMessagesSelector).length;
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please use "browser.elements"

File: features/page/loginPage.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ const login = Object.create({
 
   enterPassword: function enterPassword(queryString) {
     const searchPasswordInput = browser.element(inputPasswordLoginPage);
-    browser.pause(500);
+    browser.pause(1000);
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: it necessary?

File: features/page/homePage.js [hunk 4]
```diff
@@ -15,6 +18,22 @@ const homePage = Object.create({
 
   messagesOnPageValue: function messagesOnPageValue() {
     return browser.elements(messageCommonSelector).value.length;
+  },
+
+  inboxLeftPartUnreadMessageValue: function inboxLeftPartUnreadMessageValue() {
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please rename like `getCountUnreadMessage`

File: features/unread_inbox_amount.feature [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  In order to use GMail
+  As casual user
+  I want to see amount of unread messages next to inbox
+
+
+  Background:
+    Given Logged in to gMail home page with "akvmailer@gmail.com", "Akvelon2018!"
+
+  Scenario: Unread messages amount equal to number next to inbox
+    When I am on the home page
```
Comment (AleksandrHorev) [inline features/unread_inbox_amount.feature]: Given or Then. Not When (when it's action)

File: features/unread_inbox_amount.feature [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    Then I can see input unread messages equal to unread messages next to input
```
Comment (AleksandrHorev) [inline features/unread_inbox_amount.feature]: please rename like` I can see count unread email on label equal all unread email on pages`

File: features/page/homePage.js [hunk 7]
```diff
@@ -15,6 +18,22 @@ const homePage = Object.create({
...
+      const inboxElement = browser.elements(inboxLeftSelector).value["0"];
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: why elementS?

File: features/page/homePage.js [hunk 8]
```diff
@@ -15,6 +18,22 @@ const homePage = Object.create({
...
+      const text = browser.elementIdText(inboxElement.ELEMENT).value;
+      const numIndex = text.search('[0-9]');
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: very difficult. please check https://stackoverflow.com/questions/10003683/javascript-get-number-from-string

File: features/step_definitions/unread_inbox_abount.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+module.exports = function home(){
+    this.Given(/^Logged in to gMail home page with "(.*)", "(.*)"/, function openHomePage(loginMail, passwordMail) {
+        const gmailAddress = 'https://gmail.com';
+        browser.url(gmailAddress);
+        login.waitForVisible();
+        login.loginAs(loginMail, passwordMail);
+    });
+
+    this.When(/^I am on the home page$/, function enterEmail() {
```
Comment (AleksandrHorev) [inline features/step_definitions/unread_inbox_abount.js]: please rename `enterEmail`

File: features/step_definitions/unread_inbox_abount.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        homePage.waitForVisible();
+    });
+
+    this.Then(/^I can see input unread messages equal to unread messages next to input$/, function enterEmail() {
```
Comment (AleksandrHorev) [inline features/step_definitions/unread_inbox_abount.js]: please rename enterEmail

File: features/page/homePage.js [hunk 11]
```diff
@@ -15,6 +18,22 @@ const homePage = Object.create({
...
+      if(numIndex != -1) {
+          return Number(text.substr(numIndex, text.search('\\)') - numIndex));
+      }
+      else {
+          return 0;
+      }
+  },
+
+  inboxUnreadAmountValue: function inboxUnreadAmountValue(){
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please rename like `getCountUnreadMessage` or `getAmountUnreadMessage`

File: features/step_definitions/unread_inbox_abount.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
+const login = require('./../page/loginPage.js');
+const homePage = require('./../page/homePage.js');
+
```
Comment (AleksandrHorev) [inline features/step_definitions/unread_inbox_abount.js]: please remove 1 line

File: features/page/loginPage.js [hunk 13]
```diff
@@ -27,6 +27,14 @@ const login = Object.create({
...
     browser.waitForVisible(loginPageSelector);
   },
+
+   loginAs: function loginAs(name, password) {
+    this.enterEmail(name);
+    this.clickNextButton();
+    this.enterPassword(password);
+    this.clickNextButton();
+    this.waitForVisible();
+   }
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: Did you run lint?

File: features/step_definitions/unread_inbox_abount.js (hunk 12 above)
Comment (BigBy996) [reply features/step_definitions/unread_inbox_abount.js]: 1st line is needed to have functions login.waitForVisible and login.loginAs

Comment (AlexZavkn) [reply features/step_definitions/unread_inbox_abount.js]: I assume, Alexandr was talking about 3rd line ("one line", not "first line")

Comment (BigBy996) [reply features/step_definitions/unread_inbox_abount.js]: Thank you for clarifying! Removed it

File: features/page/homePage.js [hunk 14]
```diff
@@ -3,19 +3,38 @@ const homePageSelector = '.nH.bkL';
...
+    const inboxMessagesAmount = inboxLabelText.match(/\d+/);
+    if (inboxMessagesAmount !== null) {
+      return Number(inboxMessagesAmount['0']);
+    }
+    return 0;
+  },
+
+  getAmountUnreadMessages: function getAmountUnreadMessages() {
+    const unreadMessages = browser.elements(inboxUnreadMessagesSelector);
+    return Object.keys(unreadMessages.value).length;
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: why not `unreadMessages.value.lenght`?

**Comment by AleksandrHorev** [comment]
image

**Comment by BigBy996** [comment]
lint
passed_scenario

**Comment by BigBy996** [comment]
autotests_passed
lint

File: features/page/homePage.js [hunk 15]
```diff
@@ -14,6 +16,10 @@ const messagesRefSelector = 'td.xY.bnk.byv';
 const tableRowSelector = 'tr';
 
 const homePage = Object.create({
+  isOnHomePage: function isOnHomePage() {
+    return browser.element(homePageSelector).value !== null;
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: why not only browser.waitForVisible(homePageSelector) ?

File: features/page/homePage.js [hunk 16]
```diff
@@ -27,6 +33,21 @@ const homePage = Object.create({
     return browser.elements(messageCommonSelector).value.length;
   },
 
+  getAmountUnreadMessageInboxLabel: function getAmountUnreadMessageInboxLabel() {
+    const inboxLabel = browser.element(inboxLabelSelector).value;
+    const inboxLabelText = browser.elementIdText(inboxLabel.ELEMENT).value;
+    const inboxMessagesAmount = inboxLabelText.match(/\d+/);
```
Comment (AleksandrHorev) [inline features/page/homePage.js]: please create const for regex

File: package.json [hunk 17]
```diff
@@ -12,6 +12,7 @@
   "devDependencies": {
     "chimp": "^0.49.0",
     "eslint": "^4.16.0",
+    "eslint-config-airbnb": "*",
```
Comment (AleksandrHorev) [inline package.json]: what for it needed?

Comment (BigBy996) [reply package.json]: I received an error when executed lint (npm run lint): unable to find eslint-config-airbnb\base (or something similar)

File: features/page/homePage.js (hunk 15 above)
Comment (BigBy996) [reply features/page/homePage.js]: It seems to be working without waiting (I thouhgt that spending extra seconds for waiting is not good)
Should it be browser.waitForVisible(homePageSelector, 1) instead?

Comment (AleksandrHorev) [reply features/page/homePage.js]: You can use browser.isVisible()  http://webdriver.io/api/state/isVisible.html

**Pull request merged**